public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -496,6 +500,16 @@ export class LokiDatasource
         }
         break;
       }
+      case 'ADD_LABEL_FILTER': {
+        const parserPositions = getParserPositions(query.expr);
+        const filter = toLabelFilter('', '', '=');
+        expression = addFilterAsLabelFilter(expression, parserPositions, filter);
```

Comment:
I know that this now works, because we only offer hint if query has parser and no label fiters. But it means that this code is dependent on hints code, which isn't ideal. So here, I'd rather check for all parsers and label filters and add new label filter as last. 

This way, if we decide to use this action somewhere else/add other actions, it is going to work for any query. 

We already have all necessary functions for this such as `findLastPosition([...labelFilterPositions, ...parserPositions])`  that finds last position from all passed positions and `getParserPositions(query)` `and 
`getLabelFilterPositions(query)`.

Check the `addLabelToQuery` in `modifyQuery`